File: tests/workflow/consumer/rust/src/consumer/broadcasts/groupa_structb.rs
use super::{protocol, Consumer, Context};
use clibri::client;

pub async fn handler<E: client::Error>(
    event: protocol::GroupA::StructB,
    context: &mut Context,
    consumer: Consumer<E>,
) {
    context.broadcast.groupa_structb += 1;
    context.broadcast.check();
}

File: tests/workflow/producer/typescript (copy)/src/producer/responses/groupa.structb.ts
import { Response } from "../implementation/responses/groupa.structb";
import {
    Context,
    Producer,
    Identification,
    Filter,
    Protocol,
} from "../implementation/responses";

export function response(
    request: Protocol.GroupA.StructB,
    consumer: Identification,
    filter: Filter,
    context: Context,
    producer: Producer
): Promise<Response> {
    return Promise.reject(
    	new Error(`Handler for Protocol.GroupA.StructB isn't implemented.`)
    );
}